apollo-router/src/uplink/license_enforcement.rs

```diff
@@ -659,6 +659,15 @@ impl LicenseState {
             _ => None,
         }
     }
+
+    pub(crate) fn get_name(&self) -> String {
+        match self {
+            Self::Licensed { limits: _ } => "Licensed".to_string(),
+            Self::LicensedWarn { limits: _ } => "LicensedWarn".to_string(),
+            Self::LicensedHalt { limits: _ } => "LicensedHalt".to_string(),
+            Self::Unlicensed => "Unlicensed".to_string(),
+        }
+    }
```

Comment:
You can return a static string reference here to avoid allocation:
```suggestion
    pub(crate) fn get_name(&self) -> &'static str {
        match self {
            Self::Licensed { limits: _ } => "Licensed",
            Self::LicensedWarn { limits: _ } => "LicensedWarn",
            Self::LicensedHalt { limits: _ } => "LicensedHalt",
            Self::Unlicensed => "Unlicensed",
        }
    }
```

Reply:
oh nice, thank you!